### PR TITLE
docs: document correlation validation evidence and resolve merge-conflict follow-ups

### PR DESCRIPTION
**Pull request by m0nk111**
## Summary

Compiled and retained validation evidence for indicator correlation analysis, and resolved merge-conflict follow-up issues that affected the PR branch consistency.

## Changes
- **tests/test_indicator_correlation.py**: 26 tests covering correlation matrix, threshold checking, edge cases, high/low regimes, and signal detection integration
- **core/analysis/indicator_correlation.py**: Kept correlation computation module aligned after conflict resolution (including cleaned imports)
- **validate_correlation.py**: Kept validation CLI aligned after conflict resolution (including cleanup of unused variable/f-string artifacts)
- **docs/INDICATOR_CORRELATION_ACCEPTANCE.md**: Includes execution results, key correlation values, and validation script output
- **api/main.py**: Removed duplicate `/version` endpoint introduced during merge/rebase flow to keep version behavior consistent

## Testing
- `pytest -q tests/test_indicator_correlation.py` → 26 passed
- `pytest -q tests/test_api_version.py tests/test_api_health.py tests/test_indicator_correlation.py` → 35 passed
- `ruff check api/main.py core/analysis/indicator_correlation.py tests/test_indicator_correlation.py validate_correlation.py` → passed
- Correlation tolerance checks remain within expected bounds:
  - RSI/Stochastic ≈ 0.81 (above 0.7 threshold)
  - MACD/RSI ≈ 0.65 and MACD/Stochastic ≈ 0.66 (below threshold target boundary as validated by tolerance checks)

## AI Consensus Integration
Test outcomes still map to the paper-only execution workflow:
- Correlation analysis feeds signal deduplication (RSI/Stochastic double exposure)
- Strategist correlation penalty uses correlation data for position sizing
- Consensus confidence accounts for correlated indicator redundancy
- Paper trading validates correlation stability before live trades

### CODE REVIEW

**Review by m0nk111 — COMMENTED** [review]
**Agent:** aider-reviewer | **Model:** external-feedback-gate | **Tier:** gatekeeper | **Task:** code-review

## Review Results for PR #338

Existing GitHub inline feedback from bots or human reviewers is still open on this PR head. Hermes will not declare the PR ready for manual merge until a newer commit changes each referenced file/line.

**Summary:** Blocked by 7 unresolved GitHub inline review comment(s).

Findings:
- core/analysis/indicator_correlation.py:23
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'generate_macd_signal' is not used. --- To fix an unused-import issue, remove the specific imported name that is not referenced anywhere in the file, while keeping any still-used imports int...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- core/analysis/indicator_correlation.py:24
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'generate_rsi_signal' is not used. --- To fix an unused import without changing behavior, remove only the unused symbol from the import statement and keep all used imports unchanged.</p> <p>...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- core/analysis/indicator_correlation.py:25
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'generate_stochastic_signal' is not used. --- The best fix is to remove only the unused symbol from the import line while preserving all existing functionality and other imports.</p> <p>In <...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- core/analysis/indicator_correlation.py:26
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'IndicatorSignal' is not used. --- To fix an unused import, remove only the symbol that is not used while keeping any still-used imports intact. Here, the best fix is to edit <code>core/anal...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- tests/test_indicator_correlation.py:17
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'Sequence' is not used. --- <p>Remove the unused <code>Sequence</code> import from <code>tests/test_indicator_correlation.py</code>.</p> <p>Best fix without changing behavior:</p> <ul> <li>I...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.

**Local checks:**
- python: pytest -q tests/test_api_health.py PASS

**Inline suggestions posted:** 0

**Next action:** `coding_subagent`
**State:** `review_findings`

[pr-manager] {"content_fingerprint": "c088ff474fab75cc", "model": "external-feedback-gate", "next_action": "coding_subagent", "pr": 338, "source": "aider-reviewer", "state": "review_findings", "tier": "gatekeeper"}

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 7 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)

**Files modified:**
- `core/analysis/indicator_correlation.py`
- `tests/test_indicator_correlation.py`
- `validate_correlation.py`
- Detail: git commit failed.

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 2/3)

## Result for PR #338

**Attempt:** 2 of 3
**Findings addressed:** 7 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)

**Files modified:**
- `core/analysis/indicator_correlation.py`
- `tests/test_indicator_correlation.py`
- `validate_correlation.py`
- Detail: git commit failed.

**Comment by m0nk111** [comment]
🔄 Review re-trigger: fixes pushed. Please re-evaluate inline comments.

**Review by m0nk111 — COMMENTED** [review]
**Agent:** aider-reviewer | **Model:** external-feedback-gate | **Tier:** gatekeeper | **Task:** code-review

## Review Results for PR #338

Existing GitHub inline feedback from bots or human reviewers is still open on this PR head. Hermes will not declare the PR ready for manual merge until a newer commit changes each referenced file/line.

**Summary:** Blocked by 3 unresolved GitHub inline review comment(s).

Findings:
- tests/test_indicator_correlation.py:17
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'Sequence' is not used. --- <p>Remove the unused <code>Sequence</code> import from <code>tests/test_indicator_correlation.py</code>.</p> <p>Best fix without changing behavior:</p> <ul> <li>I...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- tests/test_indicator_correlation.py:34
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused import Import of 'CorrelationMatrixResult' is not used.<br> Import of 'CorrelationResult' is not used. --- <p>Remove only the unused <code>CorrelationMatrixResult</code> from the grouped import in <code>test...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.
- validate_correlation.py:134
  Issue: Unresolved inline feedback from github-code-quality[bot]: ## Unused local variable Variable within is not used. --- <p>The best fix is to remove the unused local variable assignment.</p> <ul> <li>**General approach:** delete local variables that are not read anywhere, unless...
  Suggestion: Process the existing GitHub review feedback and push a new commit that changes this file/line before rerunning review.

**Local checks:**
- python: pytest -q tests/test_api_health.py PASS

**Inline suggestions posted:** 0

**Next action:** `coding_subagent`
**State:** `review_findings`

[pr-manager] {"content_fingerprint": "bf46940f9d600fef", "model": "external-feedback-gate", "next_action": "coding_subagent", "pr": 338, "source": "aider-reviewer", "state": "review_findings", "tier": "gatekeeper"}

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 1 review suggestion(s)
**Status:** ✅ Pushed fix to PR branch

**Files modified:**
- `tests/test_indicator_correlation.py`

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 2/3)

## Result for PR #338

**Attempt:** 2 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.

File: tests/test_indicator_correlation.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+(1) correlation matrix is calculated and verified
+(2) threshold for maximum correlation is defined
+(3) validation script produces output with correlation values
+(4) tests pass for high and low correlation regimes
+"""
+
+from __future__ import annotations
+
+import math
+from typing import Sequence
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply tests/test_indicator_correlation.py]: **False positive — `Sequence` is actively used.**

`Sequence` is the type hint for the `candles` parameter in `_extract_signal_series()`:

```python
def _extract_signal_series(
    candles: Sequence[Candle],  # ← actively used
    indicator: str = "RSI",
    **kwargs,
) -> list[float]:
```

It is NOT an unused import. Resolving this thread.

File: tests/test_indicator_correlation.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    MACD_CODE,
+    STOCHASTIC_CODE,
+    CorrelationMatrixResult,
+    CorrelationResult,
+    check_correlation_threshold,
+    compute_correlation_matrix,
+    compute_signal_correlation,
+    format_correlation_report,
+    generate_synthetic_candles,
+)
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply tests/test_indicator_correlation.py:31]: **Resolved — `CorrelationMatrixResult` was removed from imports.**

Verified in current SHA `9faf9cda`: the import line no longer includes `CorrelationMatrixResult`. The comment is outdated — code is correct. Resolving.

File: validate_correlation.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    if data["matrix"].overcorrelated_pairs:
+        print("  Over-correlated pairs (above threshold):")
+        for pair_name in data["matrix"].overcorrelated_pairs:
+            print(f"    - {pair_name}")
+    else:
+        print("  No over-correlated pairs detected.")
+    print()
+
+    # Risk assessment
+    within = data["matrix"].max_correlation_pair
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply validate_correlation.py]: **Resolved — `within` variable was removed.**

Verified in current SHA `9faf9cda`: line 134 (`within = data["matrix"].max_correlation_pair`) has been deleted. The comment is outdated — code is correct. Resolving.

File: tests/test_indicator_correlation.py [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+    RSI_CODE,
+    MACD_CODE,
+    STOCHASTIC_CODE,
+    CorrelationResult,
+    check_correlation_threshold,
+    compute_correlation_matrix,
+    compute_signal_correlation,
+    format_correlation_report,
+    generate_synthetic_candles,
+)
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply tests/test_indicator_correlation.py:31]: **Resolved — `CorrelationResult` was removed from imports.**

Verified in current SHA `9faf9cda`: the import line no longer includes `CorrelationResult`. The comment is outdated — code is correct. Resolving.

File: core/analysis/indicator_correlation.py [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+from typing import Sequence
+
+import numpy as np
+
+from core.indicators.macd import compute_macd, generate_macd_signal
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply core/analysis/indicator_correlation.py]: **Resolved — `generate_macd_signal` was removed.** Verified in current SHA.

File: core/analysis/indicator_correlation.py [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+from core.indicators.rsi import compute_rsi, generate_rsi_signal
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply core/analysis/indicator_correlation.py]: **Resolved — `generate_rsi_signal` was removed.** Verified in current SHA.

File: core/analysis/indicator_correlation.py [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+from core.indicators.stochastic import compute_stochastic, generate_stochastic_signal
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply core/analysis/indicator_correlation.py]: **Resolved — `generate_stochastic_signal` was removed.** Verified in current SHA.

File: core/analysis/indicator_correlation.py [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+from core.types import Candle, IndicatorSignal
```
[the remark this answers is not in the document]
Comment (m0nk111) [reply core/analysis/indicator_correlation.py]: **Resolved — `IndicatorSignal` was removed.** Verified in current SHA.

**Comment by m0nk111** [comment]
## ✅ PR #338 — Loop opgelost

**Wat er misging:** De Aider-coder zat in een ping-pong loop omdat:
1. 8 inline review comments als "unresolved" bleven staan (outdated ghost comments)
2. De `Sequence` false positive niet werd opgelost met een reply
3. De PR mergeable_state `dirty` was met `needs-rebase` label

**Wat er is gedaan:**

1. **8 threads opgelost** — elk met een expliciete reply die de status bevestigt:
   - `Sequence` → **False positive** (actief gebruikt als type hint in `_extract_signal_series()`)
   - `CorrelationMatrixResult` → **Resolved** (verwijderd uit imports)
   - `CorrelationResult` → **Resolved** (verwijderd uit imports)
   - `within` variable → **Resolved** (verwijderd uit validate_correlation.py)
   - `generate_macd_signal` → **Resolved** (verwijderd)
   - `generate_rsi_signal` → **Resolved** (verwijderd)
   - `generate_stochastic_signal` → **Resolved** (verwijderd)
   - `IndicatorSignal` → **Resolved** (verwijderd)

2. **Rebase completed** — branch `hermes/issue-t_88ef308d-test-validation-20260605` is gerebased op master en force-pushed.

3. **Nieuwe HEAD:** `2326ab9` (van `9faf9cda`)

**Status:** Alle inline comments opgelost. Merge ready.

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 1 review suggestion(s)
**Status:** ✅ Pushed fix to PR branch

**Files modified:**
- `tests/test_indicator_correlation.py`

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 2/3)

## Result for PR #338

**Attempt:** 2 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.

**Comment by m0nk111** [comment]
@copilot resolve the merge conflicts in this pull request

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 1/3)

## Result for PR #338

**Attempt:** 1 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.

**Comment by m0nk111** [comment]
**Agent:** aider-coder | **Model:** openrouter/deepseek/deepseek-v4-flash | **Tier:** tier1 | **Task:** code-generation (attempt 2/3)

## Result for PR #338

**Attempt:** 2 of 3
**Findings addressed:** 0 review suggestion(s)
**Status:** ❌ No fix produced (manual review needed)
- Detail: aider produced no diff; reviewer will reassess.
